nntrainer/tensor/memory_pool.cpp

```diff
@@ -128,15 +185,28 @@ std::shared_ptr<MemoryData> MemoryPool::getMemory(unsigned int idx) {
  *
  */
 void MemoryPool::deallocate() {
+  // #ifdef ENABLE_QNN
   if (mem_pool != nullptr) {
     free(mem_pool);
     memory_size.clear();
     memory_validity.clear();
     memory_exec_order.clear();
     memory_is_wgrad.clear();
+#ifdef PROFILE
     PROFILE_MEM_DEALLOC(mem_pool);
-  }
+#endif
+    memory_ptrs.clear();
+
+#if defined(__ANDROID__)
+    int i = 0;
+    for (auto &s : memory_ptrs) {
+      if (s)
+        // allocators.at("qnn")->free(s);
+        rpcmem_free(s);
+    }
 
+#endif
```

Comment:
`memory_ptrs` should be cleared only after their elements are free.

```suggestion


#if defined(__ANDROID__)
    for (auto &s : memory_ptrs) {
      if (s)
        rpcmem_free(s);
    }
#endif
    memory_ptrs.clear();
```

Reply:
My Mistake